core/src/lib.rs

```diff
@@ -145,89 +187,101 @@ mod tests {
-        kv_subject
-            .entry("lprCategory".to_string())
-            .or_insert(Value::String("C09".to_string()));
-        kv_subject
-            .entry("lprNumber".to_string())
-            .or_insert(Value::String("999-999-999".to_string()));
-        kv_subject
-            .entry("commuterClassification".to_string())
-            .or_insert(Value::String("C1".to_string()));
-        kv_subject
-            .entry("birthCountry".to_string())
-            .or_insert(Value::String("Bahamas".to_string()));
-        kv_subject
-            .entry("birthDate".to_string())
-            .or_insert(Value::String("1958-07-17".to_string()));
+        let (kv_body, kv_subject) = get_body_subject();
 
         let vc = to.create_credential(
             crate::CRED_TYPE_PERMANENT_RESIDENT_CARD.to_string(),
             kv_subject,
             kv_body,
             "https://issuer.oidp.uscis.gov/credentials/83627465",
         );
+
         assert!(vc.is_ok());
-        assert_json_eq!(expect, vc.unwrap());
+        let credential = vc.unwrap();
+        assert_json_eq!(expect_credential, credential.serialize());
+
+        let signer = signature::signer::Ed25519DidSigner::new();
+        let proof = create_data_integrity_proof(
+            &signer,
+            credential.serialize(),
+            signature::suite::VerificationRelation::AssertionMethod,
+        );
+
+        assert!(proof.is_ok());
+
+        let verifiable_credential = credential.create_verifiable_credentials(proof.unwrap());
+        let credentials = vec![verifiable_credential];
+        let pre = to.create_presentation(credentials);
+
+        assert!(pre.is_ok());
+
+        let interim_presentation = pre.unwrap();
+
+        assert!(
+            interim_presentation.verifiable_credential[0]
+                .proof
+                .verification_method
+                .len()
+                > 0
+        );
+        assert!(
+            interim_presentation.verifiable_credential[0]
+                .proof
+                .proof_value
+                .len()
+                > 0
+        );
+
+        expect_presentation["verifiableCredential"][0]["proof"]["verification_method"] =
+            serde_json::Value::String(
+                interim_presentation.verifiable_credential[0]
+                    .proof
+                    .verification_method
+                    .clone(),
+            );
+        expect_presentation["verifiableCredential"][0]["proof"]["proof_value"] =
+            serde_json::Value::String(
+                interim_presentation.verifiable_credential[0]
+                    .proof
+                    .proof_value
+                    .clone(),
+            );
+        expect_presentation["verifiableCredential"][0]["proof"]["created"] =
+            serde_json::Value::String(
+                interim_presentation.verifiable_credential[0]
+                    .proof
+                    .created
+                    .clone(),
+            );
```

Comment:
we could use `serde_json::to_value` here to convert all of `interim_presentation.verifiable_credential[0].proof`:
```suggestion
        let interim_proof =
            serde_json::to_value(interim_presentation.verifiable_credential[0].proof.clone())
                .unwrap();
        expect_presentation["verifiableCredential"][0]["proof"] = interim_proof;
```

Reply:
Esp. since `DataIntegrityProof` already implements `serde::Serialize`
https://github.com/knox-networks/ssi/blob/9c62cff5ffb0cfce223211465b2f4ef9a112b046/core/src/proof.rs#L5-L12